lib/zinnia.js

```diff
@@ -365,8 +365,22 @@ export async function run ({
     )
     childProcesses.push(Object.assign(childProcess, { moduleName: module }))
 
+    const resetTimeout = () => {
+      if (timeoutId) clearTimeout(timeoutId)
```

Comment:
I think we need to clear the timeout after `Zinnia main loop ended`, otherwise we will end up with a leaked/danging timeout after restarting the main loop e.g. after a new module version was detected.